pkg/sdk/testint/external_functions_integration_test.go

```diff
@@ -0,0 +1,246 @@
+		require.NoError(t, err)
+		assertExternalFunction(t, id, true, true)
+	})
+
+	t.Run("alter external function: set context headers", func(t *testing.T) {
+		e := createExternalFunction(t)
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, e.Name)
+		ch := []sdk.ExternalFunctionContextHeaderRequest{
+			{
+				ContextFunction: "CURRENT_DATE",
+			},
+			{
+				ContextFunction: "CURRENT_TIMESTAMP",
+			},
+		}
+		set := sdk.NewExternalFunctionSetRequest().WithContextHeaders(ch)
+		request := sdk.NewAlterExternalFunctionRequest(id, []sdk.DataType{sdk.DataTypeVARCHAR}).WithSet(set)
+		err := client.ExternalFunctions.Alter(ctx, request)
+		require.NoError(t, err)
+		assertExternalFunction(t, id, true, true)
+	})
+
+	t.Run("alter external function: set compression", func(t *testing.T) {
+		e := createExternalFunction(t)
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, e.Name)
+		set := sdk.NewExternalFunctionSetRequest().WithCompression(sdk.String("AUTO"))
+		request := sdk.NewAlterExternalFunctionRequest(id, []sdk.DataType{sdk.DataTypeVARCHAR}).WithSet(set)
+		err := client.ExternalFunctions.Alter(ctx, request)
+		require.NoError(t, err)
+		assertExternalFunction(t, id, true, true)
+	})
+
+	t.Run("alter external function: set max batch rows", func(t *testing.T) {
+		e := createExternalFunction(t)
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, e.Name)
+		set := sdk.NewExternalFunctionSetRequest().WithMaxBatchRows(sdk.Int(20))
+		request := sdk.NewAlterExternalFunctionRequest(id, []sdk.DataType{sdk.DataTypeVARCHAR}).WithSet(set)
+		err := client.ExternalFunctions.Alter(ctx, request)
+		require.NoError(t, err)
+		assertExternalFunction(t, id, true, true)
+	})
+
+	t.Run("alter external function: unset", func(t *testing.T) {
+		e := createExternalFunction(t)
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, e.Name)
+		unset := sdk.NewExternalFunctionUnsetRequest().
+			WithComment(sdk.Bool(true)).
+			WithHeaders(sdk.Bool(true))
+		request := sdk.NewAlterExternalFunctionRequest(id, []sdk.DataType{sdk.DataTypeVARCHAR}).WithUnset(unset)
+		err := client.ExternalFunctions.Alter(ctx, request)
+		require.NoError(t, err)
+
+		assertExternalFunction(t, id, true, true)
+	})
+
+	t.Run("show external function: with like", func(t *testing.T) {
+		e1 := createExternalFunction(t)
+		e2 := createExternalFunction(t)
+
+		es, err := client.ExternalFunctions.Show(ctx, sdk.NewShowExternalFunctionRequest().WithLike(&sdk.Like{Pattern: sdk.String(e1.Name)}))
+		require.NoError(t, err)
+
+		require.Equal(t, 1, len(es))
+		require.Contains(t, es, *e1)
+		require.NotContains(t, es, *e2)
+	})
+
+	t.Run("show external function: no matches", func(t *testing.T) {
+		es, err := client.ExternalFunctions.Show(ctx, sdk.NewShowExternalFunctionRequest().WithLike(&sdk.Like{Pattern: sdk.String(random.String())}))
+		require.NoError(t, err)
+		require.Equal(t, 0, len(es))
+	})
```

Comment:
I would add a test with showById and two external functions that are created on the same database and schema and have the same name but have different args. We should be able to query the right one with identifier + args.

Reply:
added test

Reply:
I don't see it, `show external function: with like` is the closest, but those functions have different names. I would modify it so the names would be the same but the function argument types would be different. Then Show "like pattern" would probably need arguments to differentiate between those two.

Reply:
its there. this snippet here creates an external function with identifier and one argument, and verifying that the show command only returns when given proper arguments. 
```
t.Run("show external function by id", func(t *testing.T) {
		e := createExternalFunction(t, sdk.DataTypeVARCHAR)

		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, e.Name)
		es, err := client.ExternalFunctions.ShowByID(ctx, id, []sdk.DataType{sdk.DataTypeVARCHAR})
		require.NoError(t, err)
		require.Equal(t, *e, *es)

		_, err = client.ExternalFunctions.ShowByID(ctx, id, nil)
		require.Error(t, err, sdk.ErrObjectNotExistOrAuthorized)
	})
	```